packages/codegen/src/bin.ts

```diff
@@ -41,11 +45,35 @@ export async function binMain() {
       '--extension <extension>',
       'extension for newly generated files',
       '.graphql.ts'
+    )
+    .option(
+      '--resolver <resolverImportPath>',
+      'path to resolver script',
+      async (resolverImportPath) => {
+        const resolverPath = resolve(process.cwd(), resolverImportPath);
+        const module: unknown = await import(resolverPath);
+
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+        if (!module || !(module as any).default) {
+          throw new Error(
+            'module should resolve to a module that has a default export'
+          );
+        }
+
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-return
+        return (module as any).default;
+      }
     );
 
   program.parse();
 
   const options = program.opts<Options>();
 
-  await athenaCodegen(options);
+  // eslint-disable-next-line @typescript-eslint/no-misused-promises
+  let resolver = options.resolver ? await options.resolver : undefined;
```

Comment:
can we clean up the typecasting + lint disabling?

Reply:
fixed with type narrowing as you suggested offline.